cats/cats-core/src/main/java/com/netflix/spinnaker/cats/cache/Cache.java

```diff
@@ -115,4 +120,24 @@ default boolean exists(String type, String identifier) {
    * @return the items matching the type and identifiers
    */
   Collection<CacheData> getAll(String type, String... identifiers);
+
+  default Map<String, Collection<CacheData>> getAllByApplication(String type, String application) {
+    throw new InvalidCacheStateException("Method only implemented for StoreType.SQL");
```

Comment:
I don't think this exception type is representative of the actual error. I'd rather see something like `UnsupportedCacheMethodException` or something?